### DISCUSSION TITLE
failed validating the check: !(it.GetName().empty())

### DISCUSSION THREAD

**Discussion by kartikeyashivigoyal** [microsoft/onnxruntime#13278] [Other Q&A] [answered]
I have a custom-defined ONNX function in an ONNX file created with onnx_helper.make_function, which when I run in ORT fails with the error:-

onnxruntime.capi.onnxruntime_pybind11_state.Fail: [ONNXRuntimeError] : 1 : 
FAIL : Load model from test-bert-encoder-layer-23-attention-self-matmul-1-matrix-multiply-u0.onnx 
failed:ONNX Schema bert_encoder_layer_23_attention_self_matmul_1_matrix_multiply_u0: 
**failed validating the check: !(it.GetName().empty())**

But when I render it in Netron, I do see that the function has the correct name.
![image](https://user-images.githubusercontent.com/968422/195140777-c7b244b6-adc9-45ab-895c-bbbe1c2ad653.png)

What is the best way to debug such errors?

**Comment by ytaous** [comment]
Can u also check and see if all inputs/outputs have names? I think the error may come from one of those.

**Reply by kartikeyashivigoyal** [reply]
Thank you so much for the suggestion!
Saw this code:-
https://github.com/onnx/onnx/blob/f4263d8035f7f7a6af314c5f625e824b98083df9/onnx/defs/schema.cc#L757
Also, checked that the inputs and outputs were not empty and had names. There is a concept of function and node inputs, and they both match.

**Comment by kartikeyashivigoyal** [comment, marked as answer]
I was able to debug this, the issue was that the function had inputs that were not used. Here is an example:-
The graph-level view
![image](https://user-images.githubusercontent.com/968422/195395647-39879da8-4e0f-4992-b06c-46e6b1ebe350.png)
Inside the function
![image](https://user-images.githubusercontent.com/968422/195395668-a5f21f6f-5b1f-4657-93ed-c1aee6c2e3ea.png)
